contracts/oracle/VoltSystemOracle.sol

```diff
@@ -1,105 +1,161 @@
+        uint256 cachedOraclePrice = _oraclePrice; /// save a single warm SLOAD by using the stack
         uint256 timeDelta = Math.min(block.timestamp - cachedStartTime, TIMEFRAME);
-        uint256 pricePercentageChange = cachedOraclePrice * monthlyChangeRateBasisPoints / Constants.BASIS_POINTS_GRANULARITY;
+        uint256 pricePercentageChange = cachedOraclePrice * _monthlyChangeRateBasisPoints / Constants.BASIS_POINTS_GRANULARITY;
         uint256 priceDelta = pricePercentageChange * timeDelta / TIMEFRAME;
 
         return cachedOraclePrice + priceDelta;
     }
 
     /// @notice function to get the current oracle price for the OracleRef contract
-    function read()
+    /// valid is always true, price expressed as a uint256 scaled by 1e18.
+    function read() external view returns (uint256 price, bool valid) {
+        price = getCurrentOraclePrice();
+        valid = true;
+    }
+
+    /// @notice oracle price. starts off at 1e18 and compounds monthly
+    /// acts as an accumulator for interest earned in previous epochs
+    /// returns the oracle price from the end of the last period
+    function oraclePrice() external view override returns (uint256) {
+        return _oraclePrice;
+    }
+
+    /// @notice start time at which point interest will start accruing, and the
+    /// current ScalingPriceOracle price will be snapshotted and saved
+    function periodStartTime() external view override returns (uint256) {
+        return _periodStartTime;
+    }
+
+    /// @notice current amount that oracle price is inflating by monthly in basis points
+    /// does not support negative rates because PCV will not be deposited into negatively
+    /// yielding venues.
+    function monthlyChangeRateBasisPoints()
         external
         view
-        returns (Decimal.D256 memory price, bool valid)
+        override
+        returns (uint256)
     {
-        uint256 currentPrice = getCurrentOraclePrice();
-
-        price = Decimal.from(currentPrice).div(1e18);
-        valid = true;
+        return _monthlyChangeRateBasisPoints;
     }
 
     /// ------------- Public State Changing API -------------
 
     /// @notice public function that allows compounding of interest after duration has passed
     /// Sets accumulator to the current accrued interest, and then resets the timer.
     function compoundInterest() external override {
-        uint256 periodEndTime = periodStartTime + TIMEFRAME; /// save a single warm SLOAD when writing to periodStartTime
         require(
-            block.timestamp >= periodEndTime,
+            block.timestamp >= _periodStartTime + TIMEFRAME,
             "VoltSystemOracle: not past end time"
         );
 
+        _compoundInterest();
+    }
+
+    /// @notice update the change rate in basis points
+    /// callable only by the governor
+    /// when called, interest accrued is compounded and then new rate is set
+    function updateChangeRateBasisPoints(
+        uint256 newMonthlyChangeRateBasisPoints
```

Comment:
Why not set this type to be `uint16` rather than having to cast down later? Makes it clear for any integrating contract what should be passed/what the maximum value of this is